Enable WebP images in WooCommerce

anastas10s-afk opened the pull request:
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

WebP files should be able to be imported in WooCommerce, as they've been supported in WordPress core since v5.8.

Adding this pull request, based on the following comments:
- https://github.com/woocommerce/woocommerce/issues/28998#issuecomment-1131525719
- https://github.com/woocommerce/woocommerce/issues/28998#issuecomment-1128403295

This is a first for me, hope all is in order. Cheers! 🙂
PS: I am aware that line 2609 could use alignment with the lines above it; can't see that I have an option to edit the code at this point in time, though.

Closes #28998.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?
-   [ ] Have you included testing instructions?

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

anastas10s-afk committed:
Enable WebP images in WooCommerce

WebP files should be able to be imported in WooCommerce, as they've been supported in WordPress core since v5.8.

Adding this pull request, based on the following comments:
- https://github.com/woocommerce/woocommerce/issues/28998#issuecomment-1131525719
- https://github.com/woocommerce/woocommerce/issues/28998#issuecomment-1128403295

This is a first for me, hope all is in order. Cheers! 🙂

rrennick commented on plugins/woocommerce/includes/wc-core-functions.php:
```diff
@@ -2583,3 +2583,30 @@ function wc_cache_get_multiple( $keys, $group = '', $force = false ) {
+	return $mime_to_ext;
+}
+
+function wc_rest_allowed_image_mime_types() {
```
This function already exists: https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/wc-rest-functions.php#L47:L59

Adding a second definition will cause a fatal error. You need to edit the original function.

anastas10s-afk replied:
As I understand, I should go ahead with removing the whole function that I added (starting at line 2599), at wc-core-functions.php, and add `'webp'         => 'image/webp',` in line 57 at wc-rest-functions.php.

Correct?

anastas10s-afk replied:
I went ahead with doing so.

rrennick commented on plugins/woocommerce/includes/wc-core-functions.php:
```diff
@@ -2583,3 +2583,30 @@ function wc_cache_get_multiple( $keys, $group = '', $force = false ) {
+ *
+ * WebP files should be able to be imported in WooCommerce, as they've been supported in WordPress core since v5.8.
+ */
+add_filter( 'woocommerce_rest_allowed_image_mime_types', 'more_mimes_to_exts' );
```
This filter isn't necessary if you are already adding `webp` to the array being passed to the filter.

anastas10s-afk replied:
Just to clarify, it is suggested to remove line 2592. Correct?

anastas10s-afk replied:
I went ahead with doing so.

rrennick replied:
> Just to clarify, it is suggested to remove line 2592.

There is no need for the `more_mimes_to_exts` function as it was being called by the filter.

rrennick reviewed (changes requested):
@anastas10s-afk Thanks for submitting the PR. See my comments for the needed changes.

> PS: I am aware that line 2609 could use alignment with the lines above it; can't see that I have an option to edit the code at this point in time, though.

A pull request isn't restricted to a single commit. If you notice you have missed something then you can fix it in your PR branch and push it to Github. Github will automatically update your PR.

The PR template has `Closes # .` in it. The reason we have that in the template is that Github will automatically close the issue if the PR description has `Closes #ISSUENUMBER`. Can you add that to your description.

Lastly, our process requires a changelog entry. You can create one with `pnpm --filter=woocommerce changelog add`. If you don't have pnpm running in your dev environment, let me know and I can add the changelog entry for you.

anastas10s-afk commented:
> A pull request isn't restricted to a single commit. If you notice you have missed something then you can fix it in your PR branch and push it to Github. Github will automatically update your PR.

**TIL**, thank you for this @rrennick.

> The PR template has Closes # . in it. The reason we have that in the template is that Github will automatically close the issue if the PR description has Closes #ISSUENUMBER. Can you add that to your description.

Happily! Just to clarify, that would be` Closes #28998` or `Closes #37307`, though?

> Lastly, our process requires a changelog entry. You can create one with pnpm --filter=woocommerce changelog add. If you don't have pnpm running in your dev environment, let me know and I can add the changelog entry for you.

Kindly go ahead with adding the changelog entry for me, please. I do not have pnpm running in your dev environment. Thank you!

anastas10s-afk committed:
Updated WebP section in wc-core-functions.php

Removed, previously added, unnecessary filter, and already existing function in wc-rest-functions.php.

anastas10s-afk committed:
Added WebP support, at allowed_image_mime_types

rrennick commented:
> Closes #37307

@anastas10s-afk Merging a PR closes it. The `Closes #` automation only works with issues. Once you have removed the unnecessary function this will be good to go.

anastas10s-afk commented:
> > Closes #37307
> 
> @anastas10s-afk Merging a PR closes it. The `Closes #` automation only works with issues. Once you have removed the unnecessary function this will be good to go.

Noted. I went ahead with editing it to `Closes #37307.` now, thank you!

rrennick commented:
> Noted. I went ahead with editing it to Closes #37307. now, thank you!

@anastas10s-afk Sorry if I wasn't clear. Adding `Closes #37307` has no effect because this PR will be closed automatically when it is merged. `Closes #28998` needs to be in the description for Github to close the issue when the PR is merged.

> There is no need for the more_mimes_to_exts function as it was being called by the filter.

This needs to be addressed before the PR can be merged.

anastas10s-afk committed:
Removed previously added unnecessary filter

anastas10s-afk commented:
>> There is no need for the more_mimes_to_exts function as it was being called by the filter.
> 
> This needs to be addressed before the PR can be merged.

I believe everything is set now, for the PR to be merged @rrennick.

To summarize this PR, I went ahead with removing all additions I previously made in `/woocommerce/includes/wc-core-functions.php` and kept only the one at `/woocommerce/includes/wc-rest-functions.php`.

Cheers!

rapsli commented:
any timeline on when this is live?

rrennick committed:
Add changelog file

rrennick commented:
@anastas10s-afk Could you merge the latest `trunk` into this branch? We have some failing tests that were fixed in trunk.

Commit:
Merge branch 'trunk' into woocommerce-webp-support

anastas10s-afk commented:
@rrennick any pointers would be very much appreciated, if possible.

<img width="714" alt="Screenshot 2023-04-06 at 9 31 47 PM" src="https://user-images.githubusercontent.com/73900974/230465295-a2f320e6-81f8-409a-bfb2-94b254c8a852.png">

anastas10s-afk committed:
Merge remote-tracking branch 'upstream/trunk' into woocommerce-webp-support

anastas10s-afk commented:
> @anastas10s-afk Could you merge the latest `trunk` into this branch? We have some failing tests that were fixed in trunk.

Quick update on this, @rrennick. From what I gather, I was able to do so?

<img width="975" alt="Screenshot 2023-04-06 at 9 39 41 PM" src="https://user-images.githubusercontent.com/73900974/230466986-af94f172-8874-4ff2-a712-617a1b44062f.png">

rrennick commented:
> any pointers would be very much appreciated, if possible.

@anastas10s-afk I did this for you earlier today. In the future the way you would do it is update `trunk` in your fork to the latest trunk in this repo, merge trunk into your branch, then push to your origin.

anastas10s-afk commented:
> @anastas10s-afk I did this for you earlier today. In the future the way you would do it is update `trunk` in your fork to the latest trunk in this repo, merge trunk into your branch, then push to your origin.

Much appreciated, truly. A huge learning opportunity for me. Have a great one, @rrennick!
